zerver/lib/bugdown/__init__.py

```diff
@@ -1109,6 +1132,7 @@ def extendMarkdown(self, md, md_globals):
                         \*\*                         # ends by double asterisks
                        """
         md.inlinePatterns.add('stream', StreamPattern(stream_group), '>backtick')
+        md.inlinePatterns.add('tex', Tex(r'\s(?P<delim>\${1,2})(?P<body>[^$]+)(?P=delim)\s'), '>backtick')
```

Comment:
> There's only one thing that isn't working properly, and I don't know why: the `$_$ is a fun money-related smiley`! case. It's skipped (as expected) in the backend, but the frontend flags the `$_$` as invalid TeX, by painting it red. How's this possible, if both regexes are almost identical?

You have a `\s` at the beginning of your regex here, but not in the client version. Maybe since the `$_$` is at the beginning, there's no whitespace, so this doesn't match? Also, `_` is not a valid TeX math expression (since bare underscores are for subscripts), underscores could be made with `\_`.